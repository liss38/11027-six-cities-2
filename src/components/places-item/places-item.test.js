import React from 'react';
import renderer from 'react-test-renderer';
import PlacesItem from './places-item.jsx';

it(`PlacesItem renders correctly, styleMode=main`, () => {
  const tree = renderer
    .create(<PlacesItem
      offer={{
        title: `Beautiful &amp; luxurious apartment at great location`,
        type: `apartment`,
        previewImage: `img/apartment-01.jpg`,
        price: 50,
        rating: 4.5,
        isPremium: true,
        isFavorite: false,
      }}
      styleMode={`main`}
      onPlaceCardHover={jest.fn()}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`PlacesItem renders correctly, styleMode=offer`, () => {
  const tree = renderer
    .create(<PlacesItem
      offer={{
        title: `Beautiful &amp; luxurious apartment at great location`,
        type: `apartment`,
        previewImage: `img/apartment-01.jpg`,
        price: 50,
        rating: 4.5,
        isPremium: true,
        isFavorite: false,
      }}
      styleMode={`offer`}
      onPlaceCardHover={jest.fn()}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
