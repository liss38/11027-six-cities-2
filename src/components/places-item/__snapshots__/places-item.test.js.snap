// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesItem renders correctly, styleMode=main 1`] = `
<article
  className="cities__place-card place-card"
  onMouseOver={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          50
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        />
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "90%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer"
      >
        Beautiful &amp; luxurious apartment at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;

exports[`PlacesItem renders correctly, styleMode=offer 1`] = `
<article
  className="near-places__card place-card"
  onMouseOver={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          50
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        />
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "90%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer"
      >
        Beautiful &amp; luxurious apartment at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
